mm2src/coins/lightning/ln_platform.rs

```diff
@@ -0,0 +1,583 @@
+            .compat()
+            .await
+            .map_err(|e| e.into_inner())
+        {
+            Ok(bytes) => Ok(Some(deserialize(&bytes.into_vec())?)),
+            Err(err) => {
+                if let UtxoRpcError::ResponseParseError(ref json_err) = err {
+                    if let JsonRpcErrorType::Response(_, json) = &json_err.error {
+                        if let Some(message) = json["message"].as_str() {
+                            if message.contains("'code': -5") {
+                                return Ok(None);
+                            }
+                        }
+                    }
+                }
+                Err(err.into())
+            },
+        }
+    }
+
+    async fn process_tx_for_unconfirmation<T>(&self, txid: Txid, monitor: Arc<T>)
+    where
+        T: Confirm,
+    {
+        match self.get_tx_if_onchain(txid).await {
+            Ok(Some(_)) => {},
+            Ok(None) => {
+                info!(
+                    "Transaction {} is not found on chain. The transaction will be re-broadcasted.",
+                    txid,
+                );
+                monitor.transaction_unconfirmed(&txid);
+            },
+            Err(e) => error!(
+                "Error while trying to check if the transaction {} is discarded or not :{:?}",
+                txid, e
+            ),
+        }
+    }
+
+    pub async fn process_txs_unconfirmations(
+        &self,
+        chain_monitor: Arc<ChainMonitor>,
+        channel_manager: Arc<ChannelManager>,
+    ) {
+        // Retrieve channel manager transaction IDs to check the chain for un-confirmations
+        let channel_manager_relevant_txids = channel_manager.get_relevant_txids();
+        for txid in channel_manager_relevant_txids {
+            self.process_tx_for_unconfirmation(txid, channel_manager.clone()).await;
+        }
+
+        // Retrieve chain monitor transaction IDs to check the chain for un-confirmations
+        let chain_monitor_relevant_txids = chain_monitor.get_relevant_txids();
+        for txid in chain_monitor_relevant_txids {
+            self.process_tx_for_unconfirmation(txid, chain_monitor.clone()).await;
+        }
+    }
+
+    async fn get_confirmed_registered_txs(&self, client: &ElectrumClient) -> Vec<ConfirmedTransactionInfo> {
+        let registered_txs = self.registered_txs.lock().clone();
+        let mut confirmed_registered_txs = Vec::new();
+        for (txid, scripts) in registered_txs {
+            if let Some(transaction) =
+                ok_or_continue_after_sleep!(self.get_tx_if_onchain(txid).await, TRY_LOOP_INTERVAL)
+            {
+                for (_, vout) in transaction.output.iter().enumerate() {
+                    if scripts.contains(&vout.script_pubkey) {
+                        let script_hash = hex::encode(electrum_script_hash(vout.script_pubkey.as_ref()));
+                        let history = ok_or_retry_after_sleep!(
```

Comment:
I think this should be fixed in the next PR, please refer to https://github.com/KomodoPlatform/atomicDEX-API/pull/1240#discussion_r854299918
Also, I think the `history too large` error will not be returned from this since the script hashes are mostly for P2WSH addresses which are used once. I guess in the confirmation of the claiming transaction this error might appear since we claim back to our address and the script_pubkey of our address will be used in the `scripthash_get_history`, I guess I should find a solution for this but as I said this code might change and spawning these calls might be one of the solutions.

Reply:
Got it! There can be other permanent errors like invalid Electrum state etc.

Reply:
> I think a good solution for electrum errors can be to request electrums sequentially

Actually, all electrum requests are sequential through the `rpc_func!` macro, I guess I just forgot that :)